### DISCUSSION TITLE
Let a custom id be provided for a new access token

### DISCUSSION THREAD

**Discussion by gregmsanderson** [adonisjs/core#4707] [Ideas]
Hello,

I see that currently you can provide a `name` and `expiresIn` for a new access token. I think it would be useful to _also_ (optionally) let people provide `id` (which could then be a cuid or nanoid, for example).

I'm trying to keep all resource IDs consistent and since access tokens have no model, I see that the token's `id` is currently being determined from the auto-incremented primary key returned after the insert. It _looks_ like that ID is then used as part of the generated token, base64-encoded.

Essentially it would need an extra, optional id:

```js
async create(
    user: InstanceType<TokenableModel>,
    abilities: string[] = ['*'],
    options?: {
      id?: string | number | BigInt // <- ADDED
      name?: string
      expiresIn?: string | number
    }
  )
```

... in https://github.com/adonisjs/auth/blob/main/modules/access_tokens_guard/token_providers/db.ts along with some other small changes within that function to check for and use that id.

It doesn't seem worth generating a whole custom provider just to add that.

**Comment by thetutlage** [comment]
Something we would like to not expose for now. The goal is to keep things simple and avoid many moving parts.

**Reply by gregmsanderson** [reply]
Ah, ok. No problem.

I ended up making my own version of it to add in that custom id which seems to work.
